pkg/volume/portworx/portworx_util.go

```diff
@@ -151,15 +147,67 @@ func (util *PortworxVolumeUtil) UnmountVolume(u *portworxVolumeUnmounter, mountP
 	return nil
 }
 
-func (util *PortworxVolumeUtil) osdClient(hostname string) (osdvolume.VolumeDriver, error) {
-	osdEndpoint := "http://" + hostname + ":" + osdMgmtPort
+func (util *PortworxVolumeUtil) osdClient(volumeHost volume.VolumeHost) (osdvolume.VolumeDriver, error) {
 	if util.portworxClient == nil {
-		driverClient, err := volumeclient.NewDriverClient(osdEndpoint, pxdDriverName, osdDriverVersion)
-		if err != nil {
-			return nil, err
+		var e error
+
+		driverClient, err := getValidatedOsdClient(volumeHost.GetHostName())
+		if err == nil && driverClient != nil {
+			util.portworxClient = driverClient
+		} else {
+			e = err
+			kubeClient := volumeHost.GetKubeClient()
+			if kubeClient != nil {
+				nodes, err := kubeClient.CoreV1().Nodes().List(metav1.ListOptions{})
```

Comment:
Removed filtering logic in latest incremental. Please see reasoning in comment below.